File: Cpp/Largest Triangle Area.cpp
// 4ms, 93.80%
static int x = []() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);
    return 0;
}();

class Solution
{
  public:
    double largestTriangleArea(vector<vector<int>> &points)
    {
        double maxArea = 0.0;
        int pts = points.size();
        for (int i = 0; i < pts - 2; ++i)
            for (int j = i + 1; j < pts - 1; ++j)
                for (int k = j + 1; k < pts; ++k) maxArea = max(maxArea, getArea(points, i, j, k));
        return maxArea;
    }

  private:
    double getArea(vector<vector<int>> &points, int i, int j, int k)
    {
        auto &p1 = points[i], &p2 = points[j], &p3 = points[k];  // A, B, C
        return abs((p2[0] - p1[0]) * (p3[1] - p1[1]) - (p3[0] - p1[0]) * (p2[1] - p1[1]));
    }
};

File: Python3/Largest Triangle Area.py
# 236ms, 44.86%
class Solution:
    def largestTriangleArea(self, points):
        """
        :type points: List[List[int]]
        :rtype: float
        """

        def get_area(i, j, k):
            p1, p2, p3 = points[i], points[j], points[k]
            return abs((p2[0] - p1[0]) * (p3[1] - p1[1]) - (p3[0] - p1[0]) * (p2[1] - p1[1]))

        pts, max_area = len(points), 0.0
        for i in range(pts - 2):
            for j in range(i + 1, pts - 1):
                for k in range(j + 1, pts):
                    max_area = max(max_area, get_area(i, j, k))
        return 0.5 * max_area
